internal/oidc/auth/auth_handler.go

```diff
@@ -25,10 +26,16 @@ import (
 	"go.pinniped.dev/pkg/oidcclient/pkce"
 )
 
+const (
+	CustomUsernameHeaderName = "X-Pinniped-Upstream-Username"
+	CustomPasswordHeaderName = "X-Pinniped-Upstream-Password" //nolint:gosec // this is not a credential
+)
+
```

Comment:
This is to avoid unnecessary module inter-dependencies. They are defined once in the client code and once in the server code. I suppose we could make a third module for them to live, although I wasn't sure if it was worth it.

Reply:
These should live in a shared API package to signify the contract.

Reply:
Moved to #615